services/dedup/badger/badger.go

```diff
@@ -48,16 +49,22 @@ func NewBadgerDB(conf *config.Config, stats stats.Stats, path string) *Dedup {
 	badgerOpts := badger.
 		DefaultOptions(path).
 		WithCompression(options.None).
-		WithIndexCacheSize(16 << 20). // 16mb
-		WithNumGoroutines(1).
-		WithNumMemtables(conf.GetInt("BadgerDB.numMemtable", 5)).
-		WithValueThreshold(conf.GetInt64("BadgerDB.valueThreshold", 1048576)).
-		WithBlockCacheSize(0).
-		WithNumVersionsToKeep(1).
-		WithNumLevelZeroTables(conf.GetInt("BadgerDB.numLevelZeroTables", 5)).
-		WithNumLevelZeroTablesStall(conf.GetInt("BadgerDB.numLevelZeroTablesStall", 15)).
-		WithSyncWrites(conf.GetBool("BadgerDB.syncWrites", false)).
-		WithDetectConflicts(conf.GetBool("BadgerDB.detectConflicts", false))
+		WithIndexCacheSize(conf.GetInt64Var(16*bytesize.MB, 1, "BadgerDB.Dedup.indexCacheSize", "BadgerDB.indexCacheSize")).
+		WithValueLogFileSize(conf.GetInt64Var(16*bytesize.MB, 1, "BadgerDB.Dedup.valueLogFileSize", "BadgerDB.valueLogFileSize")).
+		WithBlockSize(conf.GetIntVar(int(4*bytesize.KB), 1, "BadgerDB.Dedup.blockSize", "BadgerDB.blockSize")).
+		WithMemTableSize(conf.GetInt64Var(20*bytesize.MB, 1, "BadgerDB.Dedup.memTableSize", "BadgerDB.memTableSize")).
+		WithNumMemtables(conf.GetIntVar(5, 1, "BadgerDB.Dedup.numMemtable", "BadgerDB.numMemtable")).
+		WithNumLevelZeroTables(conf.GetIntVar(5, 1, "BadgerDB.Dedup.numLevelZeroTables", "BadgerDB.numLevelZeroTables")).
+		WithNumLevelZeroTablesStall(conf.GetIntVar(10, 1, "BadgerDB.Dedup.numLevelZeroTablesStall", "BadgerDB.numLevelZeroTablesStall")).
+		WithBaseTableSize(conf.GetInt64Var(1*bytesize.MB, 1, "BadgerDB.Dedup.baseTableSize", "BadgerDB.baseTableSize")).
+		WithBaseLevelSize(conf.GetInt64Var(5*bytesize.MB, 1, "BadgerDB.Dedup.baseLevelSize", "BadgerDB.baseLevelSize")).
+		WithLevelSizeMultiplier(conf.GetIntVar(10, 1, "BadgerDB.Dedup.levelSizeMultiplier", "BadgerDB.levelSizeMultiplier")).
+		WithMaxLevels(conf.GetIntVar(7, 1, "BadgerDB.Dedup.maxLevels", "BadgerDB.maxLevels")).
+		WithNumCompactors(conf.GetIntVar(4, 1, "BadgerDB.Dedup.numCompactors", "BadgerDB.numCompactors")).
+		WithValueThreshold(conf.GetInt64Var(10*bytesize.B, 1, "BadgerDB.Dedup.valueThreshold", "BadgerDB.valueThreshold")).
+		WithSyncWrites(conf.GetBoolVar(false, "BadgerDB.Dedup.syncWrites", "BadgerDB.syncWrites")).
+		WithBlockCacheSize(conf.GetInt64Var(0, 1, "BadgerDB.Dedup.blockCacheSize", "BadgerDB.blockCacheSize")).
+		WithDetectConflicts(conf.GetBoolVar(false, "BadgerDB.Dedup.detectConflicts", "BadgerDB.detectConflicts"))
```

Comment:
Do we really need the double key? :thinking: If not you might as well use `GetBool` without `Var` as it's slightly ligher. Not a big deal though since you invoke this only once, just wondering if the double key is needed though. Do we use badger for other things but dedup?

Reply:
yes, we're using badger for reporting sampling & live events